core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorRecordSetProvider.java

```diff
@@ -13,10 +13,36 @@
  */
 package io.trino.spi.connector;
 
+import io.trino.spi.predicate.TupleDomain;
+
 import java.util.List;
 
 public interface ConnectorRecordSetProvider
 {
+    default RecordSet getRecordSet(
+            ConnectorTransactionHandle transaction,
+            ConnectorSession session,
+            ConnectorSplit split,
+            ConnectorTableHandle table,
+            List<? extends ColumnHandle> columns,
+            DynamicFilter dynamicFilter)
+    {
+        // By default, ignore dynamic filter (as it is an optimization and doesn't affect correctness).
+        return getRecordSet(transaction, session, split, table, columns, dynamicFilter.getCurrentPredicate());
+    }
+
+    default RecordSet getRecordSet(
+            ConnectorTransactionHandle transaction,
+            ConnectorSession session,
+            ConnectorSplit split,
+            ConnectorTableHandle table,
+            List<? extends ColumnHandle> columns,
+            TupleDomain<ColumnHandle> dynamicFilter)
+    {
+        // By default, ignore dynamic filter (as it is an optimization and doesn't affect correctness).
+        return getRecordSet(transaction, session, split, table, columns);
+    }
+
     default RecordSet getRecordSet(
```

Comment:
Please mark this as `@Deprecated`. Long term, we should only have the `DynamicFilter` version of this method, and remove the others.

Reply:
Sure, will do.